### ISSUE TITLE
Challenge Response Failed to connect

### ISSUE THREAD

**Issue by 641i130** [polhenarejos/pico-fido#19]
Using `ykman` in this reddit post:
https://www.reddit.com/r/KeePass/comments/opx34q/keepassxc_and_yubikeys_setting_up_the/
I'm getting the follow response from the device:
```bash
# ykman otp chalresp -t -g 2
Using a randomly generated key (hex): 4e613cf377db2a67bfeebe9ba8b70d99f90daa99
Program a challenge-response credential in slot 2? [y/N]: y
ERROR: Failed to write to the YubiKey. Make sure the device does not have restricted access (see "ykman otp --help" for more info).
```
I'm using a Xiao Seeed if it matters.

**Comment by polhenarejos** [comment]
Actually, the error comes from the same as in #12: Windows is not able to HID GET FEATURE. But whilst in #12 ykman is able to fallback to another backend, this is not the case for OTP, which only uses HID FEATURE report backend. I cannot figure out how to make it work in windows.

**Comment by 641i130** [comment]
Testing it with the newest pull, I get past the Failed to write error this time, but it seems Keepassxc doesn't detect it.
![image](https://github.com/polhenarejos/pico-fido/assets/46945263/58b758a6-faa0-4de3-baf7-974674272a3e)

![image](https://github.com/polhenarejos/pico-fido/assets/46945263/810ba59f-929d-49dd-b1ef-60b6a44fffd4)

Progress of course, love the project man, keep it up! This could become big when it does work.

**Comment by 641i130** [comment]
I'm able to get it to generate, but keepassxc doesn't seem to see it yet oddly.
![image](https://github.com/polhenarejos/pico-fido/assets/46945263/a6389fac-0ec1-42da-87d6-0fcf96f187cd)
![image](https://github.com/polhenarejos/pico-fido/assets/46945263/51aa0235-1135-45c3-befb-98f0bf08bfd3)

Heres what it should look like:
![image](https://github.com/polhenarejos/pico-fido/assets/46945263/f1ebed44-cb71-4716-a957-9d01f89d598f)

I'd like to help debug this, but I'm not sure what information is needed. Maybe USB pcap?

**Comment by polhenarejos** [comment]
Which OS?

**Comment by 641i130** [comment]
Arch Linux

**Comment by polhenarejos** [comment]
Fixed in e5d1ef2.

Do not forget to pull sdk too. It works with KeepassXC when a slot is properly configured with Challenge-response credential type (via Yubikey Manager for instance.

**Comment by 641i130** [comment]
It works! Thank you so much for making this project. It's very interesting to me, someone getting into embedded systems design! I also like that we can make our own keys instead of buying them all from Yubico!!!
